govcd/media_test.go

```diff
@@ -196,3 +239,232 @@ func (vcd *TestVCD) Test_RefreshMediaImage(check *C) {
+	itemName := `Test_DeleteMediaImage`
+
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_GetMedia: Org name not given")
+		return
+	}
+	if vcd.config.VCD.Catalog.Name == "" {
+		check.Skip("Test_GetMedia: Catalog name not given")
+		return
+	}
+
+	// Fetching organization
+	org, err := vcd.client.GetAdminOrgByName(vcd.org.Org.Name)
+	check.Assert(err, IsNil)
+	check.Assert(org, NotNil)
+
+	catalog, err := org.GetCatalogByName(vcd.config.VCD.Catalog.Name, false)
+	check.Assert(err, IsNil)
+
+	uploadTask, err := catalog.UploadMediaImage(itemName, "upload from test", vcd.config.Media.MediaPath, 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	AddToCleanupList(itemName, "mediaCatalogImage", vcd.config.VCD.Catalog.Name+"|"+vcd.vdc.Vdc.Name, "Test_UploadMediaImage")
+
+	mediaItem, err := vcd.vdc.FindMediaImage(itemName)
+	check.Assert(err, IsNil)
+	check.Assert(mediaItem, Not(Equals), MediaItem{})
+
+	task, err := mediaItem.Delete()
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	mediaItem, err = vcd.vdc.FindMediaImage(itemName)
+	check.Assert(err, IsNil)
+	check.Assert(mediaItem, Equals, MediaItem{})
+
+}
+
+func (vcd *TestVCD) Test_QueryMedia(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	skipWhenMediaPathMissing(vcd, check)
+
+	testQueryMediaName := "testQueryMedia"
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_GetMedia: Org name not given")
+		return
+	}
+	if vcd.config.VCD.Catalog.Name == "" {
+		check.Skip("Test_GetMedia: Catalog name not given")
+		return
+	}
+
+	// Fetching organization
+	org, err := vcd.client.GetAdminOrgByName(vcd.org.Org.Name)
+	check.Assert(err, IsNil)
+	check.Assert(org, NotNil)
+
+	catalog, err := org.GetCatalogByName(vcd.config.VCD.Catalog.Name, false)
+	check.Assert(err, IsNil)
+
+	uploadTask, err := catalog.UploadMediaImage(testQueryMediaName, "upload from test", vcd.config.Media.MediaPath, 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	AddToCleanupList(testQueryMediaName, "mediaCatalogImage", vcd.org.Org.Name+"|"+vcd.config.VCD.Catalog.Name, "Test_RefreshMediaImage")
+
+	media, err := catalog.QueryMedia(testQueryMediaName)
+	check.Assert(err, IsNil)
+	check.Assert(media, Not(Equals), nil)
+
+	check.Assert(media.MediaRecord.Name, Equals, testQueryMediaName)
+	check.Assert(media.MediaRecord.HREF, Not(Equals), "")
+
+	// find Invalid media
+	media, err = catalog.QueryMedia("INVALID")
+	check.Assert(err, Equals, ErrorEntityNotFound)
+	check.Assert(media, IsNil)
+}
+
+// Tests System function Refresh
+func (vcd *TestVCD) Test_RefreshMediaRecord(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	skipWhenMediaPathMissing(vcd, check)
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_GetMedia: Org name not given")
+		return
+	}
+	if vcd.config.VCD.Catalog.Name == "" {
+		check.Skip("Test_GetMedia: Catalog name not given")
```

Comment:
`Test_GetMedia` should be `TestRefreshMediaRecord`

Reply:
fixed